app/controllers/assignments_controller.rb

```diff
@@ -36,28 +36,55 @@ def index
   end
 
   def new
-    @files = Assignment.new
-    @files.course_id = @course.id
-    @files.due_date = (Time.now + 1.week).end_of_day.strftime("%Y/%m/%d %H:%M")
-    @files.available = Time.now.strftime("%Y/%m/%d %H:%M")
-    @files.lateness_config_id = @course.lateness_config_id
-    @files.request_time_taken = true
+    last_assn = @course.assignments.order(created_at: :desc).first
+    if @assignment.is_a? Files
```

Comment:
Where could this go in "the" model -- there are four distinct model classes here, and five instance variables...?